src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1680,7 +1683,8 @@ void LedgerMaster::doAdvance (ScopedLockType& sl)
             if (!standalone_ && !app_.getFeeTrack().isLoadedLocal() &&
                 (app_.getJobQueue().getJobCount(jtPUBOLDLEDGER) < 10) &&
                 (mValidLedgerSeq == mPubLedgerSeq) &&
-                (getValidatedLedgerAge() < MAX_LEDGER_AGE_ACQUIRE))
+                (getValidatedLedgerAge() < MAX_LEDGER_AGE_ACQUIRE) &&
+                (app_.getNodeStore().getWriteLoad() < MAX_WRITE_LOAD_ACQUIRE))
```

Comment:
We may want to consider the shard store as well.

```
            auto shardStore {app_.getShardStore()};
            if (!standalone_ && !app_.getFeeTrack().isLoadedLocal() &&
                (app_.getJobQueue().getJobCount(jtPUBOLDLEDGER) < 10) &&
                (mValidLedgerSeq == mPubLedgerSeq) &&
                (getValidatedLedgerAge() < MAX_LEDGER_AGE_ACQUIRE) &&
                (app_.getNodeStore().getWriteLoad() < MAX_WRITE_LOAD_ACQUIRE) &&
                (!shardStore || shardStore->getWriteLoad() < MAX_WRITE_LOAD_ACQUIRE))
```

Reply:
Ugh, this was _insane_ to read before. We need a helper function:

    bool canAcquire(...)
    {
        if (app_.getFeeTrack().isLoadedLocal())
            return false;
        ...
        if (app_.getNodeStore().getWriteLoad() >= MAX_WRITE_LOAD_ACQUIRE)
            return false;
        if (shardStore && shardStore->getWriteLoad() >= MAX_WRITE_LOAD_ACQUIRE)
            return false;
        return true;
    }